src/workerd/api/actor-state.c++

```diff
@@ -409,14 +420,19 @@ jsg::Promise<jsg::JsRef<jsg::JsValue>> DurableObjectStorageOperations::list(
   auto result = reverse
       ? getCache(OP_LIST).listReverse(kj::mv(start), kj::mv(end), limit, readOptions)
       : getCache(OP_LIST).list(kj::mv(start), kj::mv(end), limit, readOptions);
-  return transformCacheResultWithCacheStatus(js, kj::mv(result), options, &listResultsToMap);
+  return transformCacheResultWithCacheStatus(js, kj::mv(result), options, &listResultsToMap)
+      .then(js, [userSpan = kj::mv(userSpan)](jsg::Lock& js, jsg::JsRef<jsg::JsValue> value) {
```

Comment:
Let's use whenResolved instead, less error prone as it doesn't require you to re-return the value